topology/graph/indexer.go

```diff
@@ -199,12 +199,15 @@ type MetadataIndexer struct {
 // NewMetadataIndexer returns a new metadata graph indexer for the nodes
 // matching the graph filter `m`, indexing the metadata with `indexes`
 func NewMetadataIndexer(g *Graph, listenerHandler ListenerHandler, m ElementMatcher, indexes ...string) (indexer *MetadataIndexer) {
+	if len(indexes) == 0 {
+		return
```

Comment:
This will return nil and, as the callers don't check for this case, it will just segfault.
I would prefer to simply call "panic" as it would obviously be a programming error